agentendpoint/config_task.go

```diff
@@ -234,16 +242,21 @@ func (c *configTask) postCheckState(ctx context.Context) {
 	// No prepopulate run for post check as we will always check every resource.
 	for i, osPolicy := range c.Task.GetOsPolicies() {
 		ctx = clog.WithLabels(ctx, map[string]string{"config_assignment": osPolicy.GetOsPolicyAssignment(), "policy_id": osPolicy.GetId()})
-		plcy := c.policies[osPolicy.GetId()]
-
+		plcy, ok := c.policies[osPolicy.GetId()]
+		if !ok {
```

Comment:
When would this happen? We will execute every policy right?

Reply:
Normally it shouldn't, with the current flow this will always be populated. In the scenario where logic changes I want to avoid a nil pointer panic though since that will send the agent into a crash loop.